shared/src/main/scala/mlscript/helpers.scala

```diff
@@ -985,6 +987,10 @@ trait StatementImpl extends Located { self: Statement =>
   }
   
   def children: List[Located] = this match {
+    case Ann(ann, trm) => trm match {
+      case t: Ann => ann :: t.children
+      case t => ann :: t :: Nil
+    }
```

Comment:
Strange idea. Why not just `case Ann(ann, trm) => ann :: trm :: Nil`?

Reply:
Ah, I tried to keep the semantics of `Ann(Ls[Annotation], term)` and have all the annotations in one list. This isn't necessary with the updated definition of `Ann`, so I'll change this.